server/sublist.go

```diff
@@ -503,11 +503,11 @@ func (s *Sublist) Stats() *SublistStats {
 	st := &SublistStats{}
 	st.NumSubs = s.count
 	st.NumCache = uint32(len(s.cache))
-	st.NumInserts = s.inserts
-	st.NumRemoves = s.removes
-	st.NumMatches = s.matches
-	if s.matches > 0 {
-		st.CacheHitRate = float64(s.cacheHits) / float64(s.matches)
+	st.NumInserts = atomic.LoadUint64(&s.inserts)
```

Comment:
Looks like `s.inserts` and `s.removes` are updated under sublist's lock, so I don't think you need atomic for those 2.